figures/tasks.py

```diff
@@ -53,36 +58,77 @@ def populate_single_cdm(course_id, date_for=None, force_update=False):
         course_id=as_course_key(course_id)).count()
     msg = 'populate_single_cdm. course id = "{}", learner count={}'.format(
         course_id, learner_count)
-    logger.info(msg)
+    logger.debug(msg)
 
     start_time = time.time()
 
     cdm_obj, _created = CourseDailyMetricsLoader(
         course_id).load(date_for=date_for, force_update=force_update)
     elapsed_time = time.time() - start_time
-    logger.info('done. Elapsed time (seconds)={}. cdm_obj={}'.format(
+    logger.debug('done. Elapsed time (seconds)={}. cdm_obj={}'.format(
         elapsed_time, cdm_obj))
 
 
 @shared_task
-def populate_site_daily_metrics(site_id, **kwargs):
-    '''Populate a SiteDailyMetrics record
-    '''
-    logger.debug(
-        'populate_site_daily_metrics called for site_id={}'.format(site_id))
-    SiteDailyMetricsLoader().load(
-        site=Site.objects.get(id=site_id),
-        date_for=kwargs.get('date_for', None),
-        force_update=kwargs.get('force_update', False),
-        )
+def populate_single_sdm(site_id, date_for, force_update=False):
+    """Populate a SiteDailyMetrics record
+
+    This is simply a Celery task wrapper around the call to collect data into
+    the SiteDailyMetrics record for the given site and date_for.
+    """
+    logger.debug('populate_single_sdm: site_id={}'.format(site_id))
+
+    SiteDailyMetricsLoader().load(site=Site.objects.get(id=site_id),
+                                  date_for=date_for,
+                                  force_update=force_update)
+
     logger.debug(
         'done running populate_site_daily_metrics for site_id={}'.format(site_id))
 
 
+@shared_task
+def populate_daily_metrics_for_site(site_id, date_for, force_update=False):
```

Comment:
If I understand correctly, this is a task that we're now only calling as part of `populate_daily_metrics` but in theory we can run it independently later on. Correct?

Reply:
Correct. I pulled it out of `populate_daily_metrics` so that we can in the future run asynchronously (using the `.delay`) but before we do, we need to make sure that RabbitMQ can handle it without blocking other async tasks